src/Serilog/Configuration/LoggerSinkConfiguration.cs

```diff
@@ -142,8 +142,28 @@ internal LoggerSinkConfiguration(LoggerConfiguration loggerConfiguration, Action
     /// events passed through the sink.</param>
     /// <returns>Configuration object allowing method chaining.</returns>
     /// <exception cref="ArgumentNullException">When <paramref name="logger"/> is <code>null</code></exception>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public LoggerConfiguration Logger(
+        ILogger logger,
+        LogEventLevel restrictedToMinimumLevel)
+        => Logger(logger, attemptDispose: false, restrictedToMinimumLevel);
+
+    /// <summary>
+    /// Write log events to a sub-logger, where further processing may occur. Events through
+    /// the sub-logger will be constrained by filters and enriched by enrichers that are
+    /// active in the parent. A sub-logger cannot be used to log at a more verbose level, but
+    /// a less verbose level is possible.
+    /// </summary>
+    /// <param name="logger">The sub-logger.</param>
+    /// <param name="attemptDispose">Whether to shut down automatically the sub-logger
+    /// when the parent logger is disposed.</param>
+    /// <param name="restrictedToMinimumLevel">The minimum level for
+    /// events passed through the sink.</param>
+    /// <returns>Configuration object allowing method chaining.</returns>
+    /// <exception cref="ArgumentNullException">When <paramref name="logger"/> is <code>null</code></exception>
     public LoggerConfiguration Logger(
         ILogger logger,
+        bool attemptDispose = false,
         LogEventLevel restrictedToMinimumLevel = LevelAlias.Minimum)
```

Comment:
Sink configuration functions with `restrictedToMinimumLevel` should now also usually have `LevelSwitch` support; since this is a new signature, we can tack the extra parameter on without breakage before 3.0, so I'll follow up with a PR for that :-)